app-modules/interaction/src/Filament/Actions/DraftInteractionWithAiAction.php

```diff
@@ -0,0 +1,167 @@
+
+class DraftInteractionWithAiAction extends Action
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this
+            ->label('Draft with AI Assistant')
+            ->link()
+            ->icon('heroicon-m-pencil')
+            ->modalContent(fn (Page $livewire) => view('interaction::filament.actions.draft-with-ai-modal-content', [
+                'recordTitle' => $livewire->record['full_name'],
+                'avatarUrl' => AiAssistant::query()->where('is_default', true)->first()
+                    ?->getFirstTemporaryUrl(now()->addHour(), 'avatar', 'avatar-height-250px') ?: Vite::asset('resources/images/canyon-ai-headshot.jpg'),
+            ]))
+            ->modalWidth(MaxWidth::ExtraLarge)
+            ->modalSubmitActionLabel('Draft')
+            ->form([
+                Textarea::make('instructions')
+                    ->hiddenLabel()
+                    ->rows(4)
+                    ->placeholder('What do you want to write about?')
+                    ->required(),
+            ])
+            ->action(function (array $data, Get $get, Set $set, Page $livewire) {
+                $service = Feature::active('ai-integrated-assistant-settings')
+                    ? app(AiIntegratedAssistantSettings::class)->default_model->getService()
+                    : app(AiSettings::class)->default_model->getService();
+
+                $userName = auth()->user()->name;
+                $userJobTitle = auth()->user()->job_title ?? 'staff member';
+                $clientName = app(LicenseSettings::class)->data->subscription->clientName;
+                $model = $livewire::getResource()::getModelLabel();
+
+                $context = collect();
+
+                if ($id = $get('interaction_initiative_id')) {
+                    if ($initiative = InteractionInitiative::find($id)) {
+                        $context->push("- Mention the related initiative: {$initiative->name}.");
+                    }
+                }
+
+                if ($id = $get('interaction_driver_id')) {
+                    if ($driver = InteractionDriver::find($id)) {
+                        $context->push("- Include the call driver: {$driver->name}.");
+                    }
+                }
+
+                if ($id = $get('interaction_outcome_id')) {
+                    if ($outcome = InteractionOutcome::find($id)) {
+                        $context->push("- State the interaction outcome: {$outcome->name}.");
+                    }
+                }
+
+                if ($id = $get('interaction_type_id')) {
+                    if ($type = InteractionType::find($id)) {
+                        $context->push("- Specify the type of engagement: {$type->name}.");
+                    }
+                }
+
+                $additionalContext = $context->isNotEmpty() ? $context->implode("\n") : '';
+
+                try {
+                    $content = $service->complete(
```

Comment:
I would really recommend dumping this prompt and checking to see if it reads ok, because for example there is an issue on line 126 with the entire model JSON being put into the string instead of just the full name